### ISSUE TITLE
Table cell alignment ignored when using row and column span

### ISSUE THREAD

**Issue by DaelDe** [asciidoctor/asciidoctor-pdf#2352]
Hello,

consider the following table:

```adoc
[cols="^6a,^.^2,^.^3,^.^1,^.^3"]
|===
|O|R|S|i|P

.3+<|
O
2+|M
|i
|None

|centered|M|i|None
|centered|M|i|pi
|===
```

The cell with text "centered" should have center horizontal alignment. When rendered with asciidoctor-pdf 2.3.2, it has left alignment:
![grafik](https://user-images.githubusercontent.com/18593867/195063469-ceca3ca7-d6f5-4b9a-8408-6de7c9bd8b0e.png)

The table is properly rendered in asciidoclive.com, so I think this may be an issue with asciidoctor-pdf.

**Comment by mojavelinux** [comment]
I can confirm that the built-in HTML converter gets this right, whereas Asciidoctor PDF gets it wrong. Due to how Asciidoctor PDF is implemented, it may not be possible to fix it, but I'll look into it. My general advice is that when using spans (colspans and rowspans), it's best to set the alignment on the cell itself. That way, there can be no confusion, and it's a lot easier to see what is being applied to it.

> The table is properly rendered in asciidoclive.com

asciidoclive.co is not an official implementation of AsciiDoc, so it can't be used as a reference for what's correct. Though, in this case, you are right that it demonstrates the desired behavior.

**Comment by DaelDe** [comment]
Thank you for the reply and investigation of the issue.

I also tried setting the alignment on the cells only and it works as expected. The Asciidoc source looks a bit ugly when formatting is only done at cell level, but it works.

**Comment by mojavelinux** [comment]
Okay, I can explain what's happening here. The first cell with the text "centered" is actually being styled as an AsciiDoc table cell. This is due to the normal misplaced column spec inheritance in Asciidoctor cores when rowspans are used.

The reason there's a difference when compared to the HTML output is that Asciidoctor PDF does not currently support horizontal alignment on AsciiDoc table cells. This is a consequence of how AsciiDoc table cells are implemented in Asciidoctor PDF. You can observe this by removing the "a" modifier from the first column spec.

Given that most of the time non-left alignment is used when the cell's contents consists of paragraphs, I can see a way to implement this by temporarily modifying the base text alignment. I've opened a separate issue to track that feature. See #2358.

**Comment by mojavelinux** [comment]
The part we could fix is addressed by #2358 in the main branch.

Note that the misplaced column spec in general has to be dealt with in core.
